engine/include/cubos/engine/render/shader/shader.hpp

```diff
@@ -0,0 +1,31 @@
+/// @file
+/// @brief Class @ref cubos::engine::Shader.
+/// @ingroup render-shader-plugin
+
+#pragma once
+
+#include <cubos/core/gl/render_device.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Contains a shader stage created from GLSL code.
+    /// @ingroup render-shader-plugin
+    class Shader final
+    {
+    public:
+        CUBOS_REFLECT;
+
+        ~Shader() = default;
+
+        /// @brief Constructs a shader from code.
+        /// @param shaderStage Shader stage created from GLSL code.
+        Shader(cubos::core::gl::ShaderStage shaderStage)
+            : mShaderStage(std::move(shaderStage)){};
+
+        cubos::core::gl::ShaderStage getShaderStage() const;
```

Comment:
```suggestion
        /// @brief Returns the asset's shader stage.
        /// @return Shader stage.
        cubos::core::gl::ShaderStage shaderStage() const;
```

We usually omit the `get` as it's more in line with the C++ standard library styling.